src/network/message.rs

```diff
@@ -144,6 +147,35 @@ impl RawNetworkMessage {
             NetworkMessage::Alert(_)    => "alert",
         }.to_owned()
     }
+
+    /// Reads stream from a TCP socket and parses first message from it, returing
+    /// the rest of the unparsed buffer for later usage.
+    pub fn from_stream(stream: &mut Read, remaining_part: &mut Vec<u8>) -> Result<Self, encode::Error> {
+        let mut max_iterations = 16;
```

Comment:
why 16 and not 17?

Reply:
Fixed in #231 using special configuration struct `StreamReaderConfig` providing parameters for the number of iterations and buffer size. I also implemented Defaults for it with default values proportional to the powers of 2